WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -430,6 +439,32 @@ - (ReaderTopic *)findTopicNamed:(NSString *)topicName
 }
 
 /**
+ Find an existing topic with the specified topicID.
+
+ @param topicID The topicID of the topic to find in core data.
+ @return A matching `ReaderTopic` instance or nil.
+ */
+- (ReaderTopic *)findTopicWithID:(NSNumber *)topicID
+{
+    NSError *error;
+    NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:@"ReaderTopic"];
+    request.predicate = [NSPredicate predicateWithFormat:@"topicID = %@", topicID];
+
+    NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"title" ascending:YES];
+    request.sortDescriptors = @[sortDescriptor];
+    NSArray *topics = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (error) {
+        DDLogError(@"%@ error fetching topic: %@", NSStringFromSelector(_cmd), error);
+        return nil;
+    }
+
+    if ([topics count] == 0) {
+        return nil;
+    }
+    return [topics objectAtIndex:0];
```

Comment:
Actually use firstObject - it's safer to use because no error is thrown when index zero doesn't exist. :smile: